### PR TITLE
Add Queue typing

### PR DESCRIPTION
**Pull request by ooflorent**
**What kind of change does this PR introduce?**

refactoring, typing

**Did you add tests for your changes?**

no

**If relevant, link to documentation update:**

n/a

**Summary**

Add complete typing for `Queue`.

**Does this PR introduce a breaking change?**

no

**Other information**

There is an issue with generics.

### CODE REVIEW

File: lib/util/Queue.js [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
-module.exports = class Queue {
+/**
+ * @template T
+ */
+class Queue {
+	/**
+	 * @param {Iterable<T>=} items The initial elements.
+	 */
 	constructor(items) {
+		/** @type {Set<T>} */
```
Comment (mohsen1) [inline lib/util/Queue.js]: Is this private? if yes mark with `@private`

File: lib/util/Queue.js [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
 		this.set = new Set(items);
+		/** @type {Iterator<T>} */
```
Comment (mohsen1) [inline lib/util/Queue.js]: Is this private? if yes mark with `@private`

File: lib/util/Queue.js [hunk 3]
```diff
@@ -1,23 +1,46 @@
 "use strict";
 
-module.exports = class Queue {
+/**
+ * @template T
+ */
+class Queue {
+	/**
+	 * @param {Iterable<T>=} items The initial elements.
```
Comment (mohsen1) [inline lib/util/Queue.js]: You probably want `IterableIterator` which is:

```ts

interface IterableIterator<T> extends Iterator<T> {
    [Symbol.iterator](): IterableIterator<T>;
}
```

**Comment by sandersn** [comment]
Support for `@template` on classes will be in `typescript@next` starting tonight. That should fix the build errors with "Cannot find name 'T'."

**Comment by sokra** [comment]
Thanks

**Pull request merged**